astra-sim/workload/Workload.cc

```diff
@@ -103,175 +102,201 @@ void Workload::issue_dep_free_nodes() {
-      (node->getChakraNode()->node_type() == ChakraNodeType::COMM_SEND_NODE) ||
-      (node->getChakraNode()->node_type() == ChakraNodeType::COMM_RECV_NODE)) {
-    if (sys->trace_enabled) {
-      cout << "issue,sys->id=" << sys->id << ",tick=" << Sys::boostedTick()
-           << ",node->id=" << node->getChakraNode()->id()
-           << ",node->name=" << node->getChakraNode()->name() << endl;
+      issue_comm(node);
+    } else if (
+        node->type() == ChakraNodeType::INVALID_NODE) {
+      skip_invalid(node);
     }
-    issue_comm(node);
-  } else if (
-      node->getChakraNode()->node_type() == ChakraNodeType::INVALID_NODE) {
-    skip_invalid(node);
   }
 }
 
+void Workload::issue_replay(shared_ptr<Chakra::ETFeederNode> node) {
+  WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
+  wlhd->node_id = node->id();
+
+  assert(node->runtime() != 0);
+  sys->register_event(
+      this,
+      EventType::General,
+      wlhd,
+      // chakra runtimes are in microseconds and we should convert it into nanoseconds
+      node->runtime() * 1000);
+}
+
 void Workload::issue_remote_mem(shared_ptr<Chakra::ETFeederNode> node) {
   hw_resource->occupy(node);
 
   WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
   wlhd->sys_id = sys->id;
   wlhd->workload = this;
-  wlhd->node_id = node->getChakraNode()->id();
-  sys->remote_mem->issue(node->getChakraNode()->tensor_size(), wlhd);
+  wlhd->node_id = node->id();
+  sys->remote_mem->issue(node->tensor_size(), wlhd);
 }
 
 void Workload::issue_comp(shared_ptr<Chakra::ETFeederNode> node) {
-  assert(node->getChakraNode()->node_type() == ChakraNodeType::COMP_NODE);
   hw_resource->occupy(node);
 
   if (sys->roofline_enabled) {
     WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
-    wlhd->node_id = node->getChakraNode()->id();
+    wlhd->node_id = node->id();
 
     double operational_intensity =
-        static_cast<double>(node->getChakraNode()->num_ops()) /
-        static_cast<double>(node->getChakraNode()->tensor_size());
+        static_cast<double>(node->num_ops()) /
+        static_cast<double>(node->tensor_size());
     double perf = sys->roofline->get_perf(operational_intensity);
     double elapsed_time =
-        static_cast<double>(node->getChakraNode()->num_ops()) / perf;
-    uint64_t simulated_run_time =
+        static_cast<double>(node->num_ops()) / perf;
+    uint64_t runtime =
         static_cast<uint64_t>(elapsed_time * static_cast<double>(FREQ));
-    sys->register_event(this, EventType::General, wlhd, simulated_run_time);
+    sys->register_event(this, EventType::General, wlhd, runtime);
   } else {
-    WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
-    wlhd->node_id = node->getChakraNode()->id();
-
-    sys->register_event(
-        this,
-        EventType::General,
-        wlhd,
-        node->getChakraNode()->simulated_run_time());
+    issue_replay(node);
```

Comment:
@beckmabd

Reply:
OK, here is where calling the function issue_replay is a little confusing.  How about adding a comment stating "// advance this node forward the recorded "replayed" time specificed in the ET".

Reply:
Assigned to @JoongunPark.

Reply:
Resolved by @JoongunPark . @JoongunPark , please leave a comment here after you resolve a comment.